diff -Naru openfast.orig/glue-codes/fast-cpp/CMakeLists.txt openfast/glue-codes/fast-cpp/CMakeLists.txt
--- openfast.orig/glue-codes/fast-cpp/CMakeLists.txt	2018-01-24 03:34:13.193398000 +0000
+++ openfast/glue-codes/fast-cpp/CMakeLists.txt	2018-01-24 03:34:29.874604000 +0000
@@ -47,13 +47,15 @@
 add_executable(openfastcpp
   src/FAST_Prog.cpp)
 
+set(MPI_LIBRARIES "/bgsys/drivers/V1R2M4/ppc64/comm/lib/libmpichcxx-gcc.a;/bgsys/drivers/V1R2M4/ppc64/comm/lib/libmpich-gcc.a;/bgsys/drivers/V1R2M4/ppc64/comm/lib/libopa-gcc.a;/bgsys/drivers/V1R2M4/ppc64/comm/lib/libmpl-gcc.a;/bgsys/drivers/V1R2M4/ppc64/comm/lib/libpami-gcc.ndebug.a;/bgsys/drivers/V1R2M4/ppc64/spi/lib/libSPI.a;/bgsys/drivers/V1R2M4/ppc64/spi/lib/libSPI_cnk.a")
+
 target_link_libraries(openfastcpp openfastcpplib openfastlib
-  ${MPI_LIBRARIES} ${YAML_LIBRARIES} 
   ${HDF5_C_LIBRARIES}
   ${HDF5_HL_LIBRARIES}
+  ${MPI_LIBRARIES} ${YAML_LIBRARIES} 
   ${ZLIB_LIBRARIES}
   ${LIBXML2_LIBRARIES}
-  ${CMAKE_DL_LIBS})
+  ${CMAKE_DL_LIBS} pthread rt)
 
 if(MPI_COMPILE_FLAGS)
   set_target_properties(openfastcpp PROPERTIES
diff -Naru openfast.orig/modules-local/nwtc-library/src/SingPrec.f90 openfast/modules-local/nwtc-library/src/SingPrec.f90
--- openfast.orig/modules-local/nwtc-library/src/SingPrec.f90	2018-01-24 03:34:13.451171000 +0000
+++ openfast/modules-local/nwtc-library/src/SingPrec.f90	2018-01-24 03:34:51.505342000 +0000
@@ -39,9 +39,9 @@
 INTEGER, PARAMETER              :: B4Ki     = SELECTED_INT_KIND(  9 )           !< Kind for four-byte whole numbers
 INTEGER, PARAMETER              :: B8Ki     = SELECTED_INT_KIND( 18 )           !< Kind for eight-byte whole numbers
 
-INTEGER, PARAMETER              :: QuKi     = SELECTED_REAL_KIND( 20, 500 )     !< Kind for 16-byte, floating-point numbers
-INTEGER, PARAMETER              :: R8Ki     = SELECTED_REAL_KIND( 14, 300 )     !< Kind for eight-byte floating-point numbers
-INTEGER, PARAMETER              :: SiKi     = SELECTED_REAL_KIND(  6,  30 )     !< Kind for four-byte, floating-point numbers
+INTEGER, PARAMETER              :: QuKi     = 16
+INTEGER, PARAMETER              :: R8Ki     = 8
+INTEGER, PARAMETER              :: SiKi     = 4
 
 INTEGER, PARAMETER              :: BYTES_IN_SiKi =  4                           !< Number of bytes per SiKi number
 INTEGER, PARAMETER              :: BYTES_IN_R8Ki =  8                           !< Number of bytes per R8Ki number 
diff -Naru openfast.orig/modules-local/nwtc-library/src/SysGnuLinux.f90 openfast/modules-local/nwtc-library/src/SysGnuLinux.f90
--- openfast.orig/modules-local/nwtc-library/src/SysGnuLinux.f90	2018-01-24 03:34:13.451896000 +0000
+++ openfast/modules-local/nwtc-library/src/SysGnuLinux.f90	2018-01-24 03:35:10.937289000 +0000
@@ -59,13 +59,13 @@
          ! note: gamma is part of the F08 standard, but may not be implemented everywhere...
       MODULE PROCEDURE NWTC_gammaR4
       MODULE PROCEDURE NWTC_gammaR8
-      MODULE PROCEDURE NWTC_gammaR16
+      !MODULE PROCEDURE NWTC_gammaR16
    END INTERFACE
    
    INTERFACE NWTC_ERF ! Returns the ERF value of its argument
       MODULE PROCEDURE NWTC_ERFR4
       MODULE PROCEDURE NWTC_ERFR8
-      MODULE PROCEDURE NWTC_ERFR16
+      !MODULE PROCEDURE NWTC_ERFR16
    END INTERFACE
    
    
